lib/chef/chef_fs/file_system/repository/chef_repository_file_system_cookbook_dir.rb

```diff
@@ -136,7 +142,7 @@ def make_child_entry(child_name)
           end
 
           def cookbook_version
-            loader = Chef::Cookbook::CookbookVersionLoader.new(file_path, parent.chefignore)
+            loader = Chef::Cookbook::CookbookVersionLoader.new(file_path, chefignore)
```

Comment:
@vsingh-msys -- this may have changed a public API.

berkshelf definitely uses these APIs now, and I strongly suspect that at least old non-go-based chef cli in chef-dk uses them as well.  can you test that berkshelf works with chef master now?  should just be able to bundle update in the berkshelf repo and run all the features + unit tests

Reply:
Okay noted @lamont-granquist thanks!

Reply:
As I can check, berkshelf only concern about loading version cookbook with the path irrespective of the chefignore.
Ref: 
`. https://github.com/berkshelf/berkshelf/blob/master/lib/berkshelf/cached_cookbook.rb#L68
```
loader = Chef::Cookbook::CookbookVersionLoader.new(@path)
```

https://github.com/berkshelf/berkshelf/blob/master/lib/berkshelf/berksfile.rb#L181
```
loader = Chef::Cookbook::CookbookVersionLoader.new(path)
```
As class `CookbookVersionLoader` takes optional argument `chefignore` file.
https://github.com/chef/chef/blob/a16e94afeefe34abe1c57e6c4f44e25721db7082/lib/chef/cookbook/cookbook_version_loader.rb#L49

So technically no impact of chefignore changes.

Also, Have tested `berks upload`, and specs all working fine.